### PR TITLE
[Admin Module] Edit Vendor

### PR DESCRIPTION
**Pull request by AdyT14**
Completed method in controller for edit vendor, add holidays, and add overtime.

Added logic in add holidays controller method to also make changes in days off table.

### CODE REVIEW

**Comment by AdyT14** [comment]
Modified entity to use enum. I also pulled from master and merged it to my branch before these modifications. The enum values are lower case because in our local DB we all have these specific constraints for that field.

Also had problem with push. I tried to not push with what was in .idea or the other files which are not important, but it failed to push. I added them. When i will merge i will accept what is already on master concerning those files.

File: src/main/java/ro/db/vendor/domain/HolidayType.java [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+      default:
+        throw new IllegalArgumentException("This holidayType not supported");
+    }
+  }
+
+  public String getType() {
+    return this.type;
+  }
+
+}
```
Comment (alexandrubirlan) [inline src/main/java/ro/db/vendor/domain/HolidayType.java]: good job

File: src/main/java/ro/test/service/HolidayServiceImplTest.java [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+import ro.db.vendor.service.HolidayService;
+import ro.db.vendor.service.VendorService;
+
+@RunWith(value = SpringJUnit4ClassRunner.class)
+@ContextConfiguration(classes = PersistenceConfig.class)
+@Transactional
+public class HolidayServiceImplTest {
+
+  @Inject
+  private HolidayService holidayService;
```
Comment (alexandrubirlan) [inline src/main/java/ro/test/service/HolidayServiceImplTest.java]: unless this comes from a separate test Spring context, it should be mocked;
do not use real database in unit tests

File: src/main/java/ro/test/service/OvertimeServiceImplTest.java [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+import ro.db.vendor.service.OvertimeService;
+import ro.db.vendor.service.VendorService;
+
+@RunWith(value = SpringJUnit4ClassRunner.class)
+@ContextConfiguration(classes = PersistenceConfig.class)
+@Transactional
+public class OvertimeServiceImplTest {
+
+  @Inject
+  private OvertimeService overtimeService;
```
Comment (alexandrubirlan) [inline src/main/java/ro/test/service/OvertimeServiceImplTest.java]: same as above; mock or use a mock database

**Pull request merged**